build_model_tree.js

```diff
@@ -0,0 +1,127 @@
+const fs = require("fs");
+const path = require("path");
+const yaml = require("js-yaml");
+
+const ASSET_PATH = path.resolve(__dirname, "assets");
+
+/**
+ * Load YAML asset files and insert asset data into the MODEL_DATA object.
+ * @todo expand functionality of this function if more sophisticated file loading is needed,
+ *       e.g. loading and following references
+ * @param {string} pathToAsset - asset file path
+ */
+const loadAssetFile = (pathToAsset) => {
+    const fileContent = fs.readFileSync(pathToAsset, "utf8");
+    const yamlObj = yaml.load(fileContent);
+    // process data
+    return yamlObj;
+};
+
+const getDirectories = (currentPath) => {
+    return fs
+        .readdirSync(currentPath, { withFileTypes: true })
+        .filter((dirent) => dirent.isDirectory())
+        .map((dirent) => dirent.name);
+};
+
+const getAssetFiles = (currentPath, assetExtension = ".yml") => {
+    return fs
+        .readdirSync(currentPath)
+        .filter((dirItem) => path.extname(dirItem) === assetExtension);
+};
+
+/**
+ * Check whether asset file is an auxiliary file.
```

Comment:
What is Auxiliary file?

Reply:
Per our discussion, I moved the auxiliary files to a dedicated directory at the top level: `manifest`